src/main/scala/io/iohk/ethereum/ledger/BlockExecution.scala

```diff
@@ -59,7 +62,9 @@ class BlockExecution(
         .getBlockHeaderByHash(block.header.parentHash)
         .toRight(MissingParentError) // Should not never occur because validated earlier
       execResult <- executeBlockTransactions(block, parent)
-      worldToPersist = blockPreparator.payBlockReward(block, execResult.worldState)
+      worldToPersist <- Try {
+        blockPreparator.payBlockReward(block, execResult.worldState)
+      }.toEither.left.map(BlockExecutionError.MPTError(_))
```

Comment:
with Cats this could be rewritten to
```
Either
  .catchOnly[Throwable](blockPreparator.payBlockReward(block, execResult.worldState))
  .leftMap(BlockExecutionError.MPTError.apply)
```

Reply:
Ooh nice!